### PR TITLE
master/f22-branch - Handle New_Repository name collision source spoke (#1125322)

### PR DESCRIPTION
**Pull request by bcl**
When adding more addon repos without editing the name they would all be
called 'New_Repository". This will append a unique number to the end of
the name if the name already exists in the repoStore.

### CODE REVIEW

File: pyanaconda/ui/gui/spokes/source.py [hunk 1]
```diff
@@ -1174,6 +1174,45 @@ def _reset_repoStore(self):
...
+
+        matches = sorted((r[REPO_NAME_COL] for r in self._repoStore \
+                          if name_re.match(r[REPO_NAME_COL])), key=key)
+        if not matches:
+            return name+"_1"
+
+        # Get the highest number, add 1, append to name
+        name_match = name_re.match(matches[-1])
+        next_d = int(name_match.group(2)) + 1
+        return name_match.group(1)+"_%d" % next_d
```
Comment (vpodzime) [inline pyanaconda/ui/gui/spokes/source.py]: Wow, this whole thing looks really complicated for something that at first glance seems like an easy thing to do. I guess that's necessary to have it solid enough however, did you think about using a dict to store the "name: highest_index" pairs that would make the code a bit simpler?

**Comment by bcl** [comment]
Sorry, I'm not sure what you mean. Can you elaborate?

**Comment by vpodzime** [comment]
What I meant was to use a "repo_name:highest_index" mapping (dict) to facilitate the parsing code and deriving the next index. Once a repo name would be parsed, the mapping would be checked for the current highest index and +1 would be used (or "1" if not found in the mapping).

**Comment by bcl** [comment]
Ah, ok, I think I see. I don't think that would help much, just spread the complexity around. Right now it's all based on the repo name strings, which can come from hitting +, users naming things or kickstart so I think it's better to just try and parse those instead of converting them to additional storage.

**Comment by vpodzime** [comment]
Fair enough. It would be great if we could make the new code a little bit more independent on the GUI it is part of so that we could write tests for it more easily, but ACK from me on this change even as it is.

**Pull request merged**